crates/rpc/rpc/src/eth/api/call.rs

```diff
@@ -222,11 +227,14 @@ where
         // transaction requires to succeed
         let gas_used = res.result.gas_used();
         // the lowest value is capped by the gas it takes for a transfer
-        let mut lowest_gas_limit = MIN_TRANSACTION_GAS;
+        let mut lowest_gas_limit =
+            if (env.tx.transact_to.is_create()) { MIN_CREATE_GAS } else { MIN_TRANSACTION_GAS };
         let mut highest_gas_limit: u64 = highest_gas_limit.try_into().unwrap_or(u64::MAX);
         // pick a point that's close to the estimated gas
-        let mut mid_gas_limit =
-            std::cmp::min(gas_used * 3, (highest_gas_limit + lowest_gas_limit) / 2);
+        let mut mid_gas_limit = std::cmp::min(
+            gas_used * 3,
+            ((highest_gas_limit as u128 + lowest_gas_limit as u128) / 2) as u64,
+        );
```

Comment:
why is this better?

Reply:
It accounts for possible overflow when `highest_gas_limit + lowest_gas_limit` is bigger than `u64::MAX`. This happened with the `CallRequest` used in `test_basic_eth_calls`. However, it could also be approximated by dividing both numbers before adding them (this would result in a maximum error of 2 units).